### DISCUSSION TITLE
Forwarder sequental query order ?

### DISCUSSION THREAD

**Discussion by c-michail** [TechnitiumSoftware/DnsServer#993] [General]
Hello,
  Is is possible to force the DNS Server to query the forwarders at sequential order ?
  I mean to always query from the first and if it fails to query from the 2nd and so on ...

Thank you

**Comment by Hemsby** [comment]
If im not wrong, I think if you change Forwarder Concurrency value to 1 this would then achieve that.

![image](https://github.com/user-attachments/assets/aab2d9f8-1adc-4851-b05d-dd988fa9aff0)

**Reply by c-michail** [reply]
It's already 1 and queries from secondary also even if the primary is online and active.

Here's my setthings
![εικόνα](https://github.com/user-attachments/assets/5f8d824a-b2cf-42ee-9c97-553ac378a566)

I am checking it from NextDNS. It reports Google's DNS while my primary is Cloudflare on Technitium.

![εικόνα](https://github.com/user-attachments/assets/9f5fa5ef-396f-4ad4-a165-3986dc71d569)

**Reply by c-michail** [reply]
Well DNS Leak test finds all my forwarders with the above settings

![εικόνα](https://github.com/user-attachments/assets/fe3f703d-4a09-47b7-9d3f-8d56bf5660cf)

**Comment by ShreyasZare** [comment]
Thanks for asking. Currently there is no option to do that but in the next update, you can create conditional forwarder root zone and add multiple FWD records with `Priority` of the order they should be used.

![image](https://github.com/user-attachments/assets/edba353c-2b3f-4e1f-b6b5-dfba0b686088)

**Reply by c-michail** [reply]
Well it will be much better to have a "Sequential Order" check box in the forwarder section like some routers have to do the same instead of creating conditional forwarder. Any how it shouldn't be that difficult. The priority should be set from the list. Eg. always try the FIRST forwarder and if timeouts or gets error try the 2nd and so on ... 

Thank you

**Reply by ShreyasZare** [reply]
Thanks for the feedback. Will evaluate this option.

**Reply by c-michail** [reply]
Here's a algo just from my head: 

Forworders Count = 3;
ForworderIndex =1;

do sequental loop
  Query;

  if Query = Timeout or Forwarder Fail then
   ForworderIndex := ForworderIndex + 1 //sets the 2nd forworder
 else 
   exit; //all fine from the first forwarde so exit the loop
until  ForworderIndex > Forworders Count;

if ForworderIndex > Forworders Count 
{
  Server Error All Forwarders failed or no internet connection
}


Another idea is that if the First forwarder fails switch to the next working forwarder available for some minutes (interval which we can set) and the try back the first to see if it's available and switch back to the first. In that case you may skip the loops above.

**Reply by ShreyasZare** [reply]
Will get this added in the next update.

**Reply by c-michail** [reply]
btw Dnsmasq has something similar with the option ‘strict-order’
